### DISCUSSION TITLE
Typed `Outlet` `context`

### DISCUSSION THREAD

**Discussion by TranquilMarmot** [remix-run/remix#5103] [Q&A] [answered]
I'm trying to figure the best way to type the `context` that's passed to an `Outlet`.

Right now, I have something like...

```tsx
interface ParentComponentOutletContext {
   something?: string;
}

const ParentComponent = () => {
   // `context` here is untyped!
   return (
      <Outlet context={{ something: 'hello' }} />
   );
}

// ... in the child route ...

const ChildComponent = () => {
   const { something } = useOutletContext<ParentComponentOutletContext>();

   return (
      <div>{something}</div>
   )
}
```

This works alright, but if I want to change the type of `ParentComponentOutletContext` to add something else, but I forget to add it to the `Outlet`'s `context`, there should be a type error but there isn't.

Since `context` is always an `unknown`, there's effectively no type-checking going on when rendering the JSX.

I know there isn't really a way to pass a type parameter in JSX, but I'm still wondering if there's a better way to handle this.

**Comment by sergiodxa** [comment, marked as answer]
You can pass generics to JSX by doing this `<ComponentName<GenericType> prop />` so it’s possible to make Outlet accept a generic 

But I doubt that will solve your use case, because Outlet context accepts any value and because you can have multiple levels of nested routes each rendering an Outlet with a different value there’s no way to ensure the value returned by useOutletContext is what the supposed parent route says, remember a component could call the hook and be rendered in multiple routes.

If you want to be sure your component receives what it expects it’s better to check in runtime.
